tokio/src/stream/empty.rs

```diff
@@ -10,6 +10,8 @@ use core::task::{Context, Poll};
 pub struct Empty<T>(PhantomData<T>);
 
 impl<T> Unpin for Empty<T> {}
+unsafe impl<T> Send for Empty<T> {}
```

Comment:
This manual impl *should* be unnecessary if `Empty<T>` held a `PhantomData<fn() -> T>` instead. (I think that's the right `PhantomData` for this case).

Reply:
Yes, I believe you are correct. Changing the `PhantomData` should be enough and avoid the `unsafe` impl.

Reply:
It isn't possible to do that because function pointers in const fns are unstable, and the constructor is a const fn. I can use `std::iter::Empty<T>` instead though.